subsystems/visionsubsystem.py

```diff
@@ -117,6 +119,106 @@ def reset(self) -> None:
         pass
 
 
+class BallTrackingModule:
+    def __init__(self) -> None:
+        self.targetAngle = None
+        self.targetDistance = None
+
+    def getTargetAngle(self) -> typing.Optional[Rotation2d]:
+        return self.targetAngle
+
+    def getTargetDistance(self) -> typing.Optional[float]:
+        return self.targetDistance
+
+    def update(self) -> None:
+        NetworkTables.initialize()
+        self.photonvisionNetworkTable = NetworkTables.getTable(
+            constants.kPhotonvisionNetworkTableName
+        )
+
+        if self.photonvisionNetworkTable.getBoolean(
+            constants.kPhotonvisionTargetValidKey, False
+        ):
+            if RobotBase.isReal():
+                ballAngleYToCamera = Rotation2d.fromDegrees(
+                    self.photonvisionNetworkTable.getValue(
+                        constants.kPhotonvisionTargetVerticalAngleKey, 0
+                    )
+                )
+
+                distanceToCamera = (
+                    Rotation2d.tan(
+                        ballAngleYToCamera.radians()
+                        + constants.kIntakeCameraMaxVerticalFOV.radians() / 2
+                    )
+                    * constants.kIntakeCameraHeightInMeters
+                )
+            else:
+                distanceToCamera = self.photonvisionNetworkTable.getValue(
+                    constants.kPhotonvisionTargetSimDistanceKey, float("inf")
+                )
+
+            if not constants.kIsIntakeCameraCentered:
```

Comment:
given current design I think it should be safe to say that the camera WILL be centered horizontally (allowing the largest possible space on each side to see)
in the event that it isnt the offset should make is such that the differences zero out and result in no need for this if statement